pkg/daemon/ceph/osd/agent.go

```diff
@@ -654,10 +654,12 @@ func isBluestore(config *osdConfig) bool {
 }
 
 func isBluestoreDevice(cfg *osdConfig) bool {
-	return !cfg.dir && cfg.partitionScheme != nil && cfg.partitionScheme.StoreType == config.Bluestore
```

Comment:
a bit paranoid but i would validate `cfg != nil` first

Reply:
For public methods we would want to validate. For private methods I'd say it is not a worry.